provider/internal/builtin/service_client_test.go

```diff
@@ -693,6 +693,72 @@ func Test_builtinServiceClient_Evaluate_InclusionExclusion(t *testing.T) {
 			},
 			wantFilePaths: []string{filepath.Join("dir_b", "b.xml")},
 		},
+		{
+			name:       "(Filecontent) Include files from cond.Filepaths (single rendered path), no include / exclude",
+			capability: "filecontent",
+			condition: builtinCondition{
+				Filecontent: fileContentCondition{
+					Pattern: "(fox|app.config.property = .*)",
+					Filepaths: []string{fmt.Sprintf("%s %s",
+						filepath.Join(baseLocation, "dir_a", "a.txt"), filepath.Join(baseLocation, "dir_b", "b.txt")),
+					},
+				},
+			},
+			chainTemplate: engine.ChainTemplate{},
+			wantFilePaths: []string{
+				filepath.Join("dir_a", "a.txt"),
+				filepath.Join("dir_b", "b.txt"),
+			},
+		},
+		{
+			name:       "(Filecontent) Include files from cond.Filepaths (single rendered path), with include",
+			capability: "filecontent",
+			condition: builtinCondition{
+				Filecontent: fileContentCondition{
+					Pattern: "(fox|app.config.property = .*)",
+					Filepaths: []string{fmt.Sprintf("%s %s",
+						filepath.Join(baseLocation, "dir_a", "a.txt"), filepath.Join(baseLocation, "dir_b", "b.txt")),
+					},
+				},
+			},
+			chainTemplate: engine.ChainTemplate{
+				Filepaths: []string{
+					// here we test whether intersection works as expected
```

Comment:
I am a little confused by this intersection bit here, and think that will need to document this for rule authors. 

IIUC, this would mean that anytime I use a From/As block, then the file paths are automatically set, to the intersection of what I have provided as the rule author and the output of the <As> conditions file paths? 

This seems like a pretty big breaking change, and if there is no way to turn it off, then I think we need to rework it.

Reply:
I do think that we need to document this, lets just make sure we add that to the chaining documentation?

Reply:
e1a1251